packages/clerk_flutter/lib/src/widgets/ui/clerk_input_dialog.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:clerk_flutter/clerk_flutter.dart';
+import 'package:flutter/material.dart';
+
+/// Creates a dialog containing a [child] widget which, typically,
+/// allows some kind of input
+///
+class ClerkInputDialog extends StatelessWidget {
+  /// Construct a [ClerkInputDialog]
+  const ClerkInputDialog._({
+    required this.child,
+    required this.auth,
+    this.showOk = true,
+  });
+
+  /// The child [Widget]
+  final Widget child;
+
+  /// An injected [ClerkAuthProvider]
+  final ClerkAuthProvider auth;
+
+  /// Show the OK button
+  final bool showOk;
+
+  /// Display an input box containing [child]
+  static Future<bool> show(
+    BuildContext context, {
+    required Widget child,
+    bool showOk = true,
+  }) async {
+    final auth = ClerkAuth.of(context, listen: false);
+    return await showDialog(
+      context: context,
+      builder: (_) => ClerkInputDialog._(
+        auth: auth,
+        showOk: showOk,
+        child: child,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final translator = auth.translator;
+    return AlertDialog(
```

Comment:
Use adaptive dialog

Reply:
@slightfoot Are you 100% certain about using the adaptive_dialog package? It's another dependency, it's a nightmare to work with, and it looks hideous.

Reply:
no we won't use a package for this. there are other solutions.